test/planner/plan_util_test.cpp

```diff
@@ -0,0 +1,170 @@
+  // Insert a table first
+  auto id_column = catalog::Column(
+      type::TypeId::INTEGER, type::Type::GetTypeSize(type::TypeId::INTEGER),
+      "id", true);
+  auto fname_column =
+      catalog::Column(type::TypeId::VARCHAR, 32, "first_name", false);
+  auto lname_column =
+      catalog::Column(type::TypeId::VARCHAR, 32, "last_name", false);
+
+  std::unique_ptr<catalog::Schema> table_schema(
+      new catalog::Schema({id_column, fname_column, lname_column}));
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  catalog->CreateTable(TEST_DB_NAME, "test_table", std::move(table_schema),
+                       txn);
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  auto source_table = db->GetTableWithName("test_table");
+  oid_t col_id = source_table->GetSchema()->GetColumnID(id_column.column_name);
+  LOG_INFO("id: %d", static_cast<int>(col_id));
+  std::vector<oid_t> source_col_ids;
+  source_col_ids.push_back(col_id);
+
+  // create index on 'id'
+  catalog->CreateIndex(TEST_DB_NAME, "test_table", source_col_ids,
+                       "test_id_idx", false, IndexType::BWTREE, txn);
+
+  // create index on 'id' and 'first_name'
+  col_id = source_table->GetSchema()->GetColumnID(fname_column.column_name);
+  LOG_INFO("fname: %d", static_cast<int>(col_id));
+  source_col_ids.push_back(col_id);
+
+  catalog->CreateIndex(TEST_DB_NAME, "test_table", source_col_ids,
+                       "test_fname_idx", false, IndexType::BWTREE, txn);
+  txn_manager.CommitTransaction(txn);
+
+  // dummy txn to get the catalog_cache object
+  txn = txn_manager.BeginTransaction();
+
+  // This is also required so that database objects are cached
+  auto db_object = catalog->GetDatabaseObject(TEST_DB_NAME, txn);
+  EXPECT_EQ(static_cast<int>(db_object->GetTableObjects().size()), 1);
+
+  // Till now, we have a table : id, first_name, last_name
+  // And two indexes on following columns:
+  // 1) id
+  // 2) id and first_name
+  auto table_object = db_object->GetTableObject("test_table");
+  oid_t id_idx_oid = table_object->GetIndexObject("test_id_idx")->GetIndexOid();
+  oid_t fname_idx_oid =
+      table_object->GetIndexObject("test_fname_idx")->GetIndexOid();
+
+  // An update query affecting both indexes
+  std::string query_string = "UPDATE test_table SET id = 0;";
+  std::unique_ptr<Statement> stmt(new Statement("UPDATE", query_string));
+  auto &peloton_parser = parser::PostgresParser::GetInstance();
+  auto sql_stmt_list = peloton_parser.BuildParseTree(query_string);
+  auto sql_stmt = sql_stmt_list->GetStatement(0);
+  static_cast<parser::UpdateStatement *>(sql_stmt)->table->TryBindDatabaseName(
+      TEST_DB_NAME);
+  std::set<oid_t> affected_indexes =
+      planner::PlanUtil::GetAffectedIndexes(txn->catalog_cache, *sql_stmt);
+
+  // id and first_name are affected
+  EXPECT_EQ(static_cast<int>(affected_indexes.size()), 2);
```

Comment:
For these assertions/expectations, the expected value should be the first argument.